build.gradle

```
@@ -51,62 +45,18 @@ compileJava {
 }
 
 jar {
-    archiveBaseName = 'jextract'
+    archiveBaseName = 'org.openjdk.jextract'
     archiveVersion = project.version
 }
 
-moditect {
-    // moditect uses java.home to find JDK tools
-    System.setProperty("java.home", "$jdk18_home")
-
-    // inject module-info.class for org.openjdk.jextract module
-    addMainModuleInfo {
-        dependsOn build
-
-        jvmVersion = '18'
-        version = project.version
-        module {
-            moduleInfoSource = '''
-                module org.openjdk.jextract {
-                    requires transitive java.compiler;
-                    requires transitive jdk.incubator.foreign;
-                    requires jopt.simple;
-                    requires java.prefs;
-                    exports org.openjdk.jextract;
-
-                    provides java.util.spi.ToolProvider with
-                       org.openjdk.jextract.JextractTool.JextractToolProvider;
-                }
-                '''
-        }
-    }
-
-    // inject module-info.class for jopt.simple module
-    addDependenciesModuleInfo {
-        jvmVersion = '18'
-        outputDirectory = file("$buildDir/modules")
-        modules {
-            module {
-                artifact 'net.sf.jopt-simple:jopt-simple:5.0.4'
-                moduleInfoSource = '''
-                    module jopt.simple {
-                        exports joptsimple;
-                        exports joptsimple.util;
-                    }
-                '''
-            }
-        }
-    }
-}
-
 task createJextractImage(type: Exec) {
-    dependsOn addMainModuleInfo, addDependenciesModuleInfo
+    dependsOn build
```

```suggestion
    dependsOn jar
```
Depending on `jar` is more correct here as what you require is the compiled jar in `build/libs`, and not the completion of the build lifecycle task.

---

Thanks for catching this.